src/core/notation.js

```diff
@@ -91,29 +91,45 @@ class Notation {
      *  // "car.model"  "Charger"
      *  // "car.year"  1970
      */
-    each(callback) {
-        let o = this._source,
-            keys = Object.keys(o);
+    each (callback) {
```

Comment:
This yields keys such as `a[0]` instead of `[0]` which produces bugs. Pls see new tests I've attached to the PR conversation.